apps/ltn/src/logic/turn_restrictions.rs

```diff
@@ -0,0 +1,347 @@
+    let road = map.get_r(r);
+    return road.dst_i == i || road.src_i == i
+}
+
+/// Returns a HashSet of all roads which are connected by driving from RoadID.
+/// This accounts for oneway restrictions, but not turn restrictions. eg:
+/// 
+/// - If a oneway restriction on either the 'from_road' or the 'target_road' would prevent driving from
+/// source to destination, then 'target_road' it will NOT be included in the result.
+/// - If a turn restriction exists and is the only thing that would prevent driving from 'from_road' or the
+/// 'target_road', then the 'target_road' will still be included in the result.
+/// 
+/// `i` is Optional. If `i` is `Some` then, it must be connected to `from_r`. It is used to filter
+/// the results to return only the destination roads that connect to `i`.
+/// 
+// TODO highlighting possible destinations for complicated turns (at present both sections of existing
+// complicated_turn_restrictions are included). However possible future complicated turns are not detected.
+//
+// TODO Rework `possible_destination_roads()` and `restricted_destination_roads()` to a extra function that
+// returns a tupple `(permitted, opposing_oneway, restricted_turn)`
+pub fn possible_destination_roads(map: &Map, from_r: RoadID, i: Option<IntersectionID>) -> HashSet<RoadID> {
+
+    if i.is_some() && !verify_intersection(map, from_r, i.unwrap()){
+        panic!("IntersectionID {:?}, does not connect to RoadID {:?}", i, from_r);
+    }
+
+    let from_road = map.get_r(from_r);
+    let mut target_roads: HashSet<RoadID> = HashSet::new();
+
+    let one_way = from_road.oneway_for_driving();
+
+    if one_way != Some(Direction::Fwd) && Some(from_road.dst_i) != i {
+        for r in &map.get_i(from_road.src_i).roads {
+            if from_road.id != *r && is_road_drivable_from_i(&map, *r, from_road.src_i){
+                target_roads.insert(*r);
+            }
+        }
+    }
+
+    if one_way != Some(Direction::Back) && Some(from_road.src_i) != i {
+        for r in &map.get_i(from_road.dst_i).roads {
+            if from_road.id != *r && is_road_drivable_from_i(&map, *r, from_road.dst_i) {
+                target_roads.insert(*r);
+            }
+        }
+    }
+    target_roads
+}
+
+fn is_road_drivable_from_i(map: &Map, target_r: RoadID, i: IntersectionID) -> bool {
+
+    let road = map.get_r(target_r);
+    let one_way = road.oneway_for_driving();
+    
+    return road.is_driveable() &&
+           (road.src_i == i && one_way != Some(Direction::Back)) ||
+           (road.dst_i == i && one_way != Some(Direction::Fwd)) 
+}
+
+#[cfg(test)]
+mod tests {
+    use tests::{import_map, get_test_file_path};
+    use super::{possible_destination_roads, restricted_destination_roads, FocusedTurns};
+    use map_model::{RoadID, IntersectionID};
+    use std::collections::HashSet;
+    use geom::Pt2D;
+
+    #[test]
+    fn test_focused_turn_restriction() -> Result<(), anyhow::Error> {
```

Comment:
It's not really typical for Rust unit tests to return a `Result`. It could be more convenient in places where you end up using `?` a bunch and don't want to `unwrap` for some reason. In this case, I think you could just leave off the return type

Reply:
Done